app/llm/performance_enhanced.py

```
@@ -0,0 +1,142 @@
+    """
+    Thread-safe LRU cache manager with performance tracking.
+    """
+
+    def __init__(self, max_size: int = 1000):
+        self._cache: OrderedDict[str, Any] = OrderedDict()
+        self._max_size = max_size
+        self._hits = 0
+        self._misses = 0
+        self._lock = threading.Lock()
+
+    def get(self, key: str) -> Optional[Any]:
```

Ruff (0.8.2)

Dynamically typed expressions (typing.Any) are disallowed in `get`

(ANN401)